pkg/loop/internal/relayerset/contract_reader.go

```diff
@@ -124,65 +107,109 @@ func (s *Server) ContractReaderClose(_ context.Context, req *relayerset.Contract
 	return &emptypb.Empty{}, reader.reader.Close()
 }
 
-func (s *Server) ContractReaderGetLatestValue(ctx context.Context, req *relayerset.ContractReaderGetLatestValueRequest) (*pb.GetLatestValueReply, error) {
-	reader, err := s.getReader(req.ContractReaderId)
+func (s *Server) GetLatestValue(ctx context.Context, in *pb.GetLatestValueRequest) (*pb.GetLatestValueReply, error) {
+	readerId, err := getContractReaderId(ctx)
+	if err != nil {
+		return nil, err
+	}
+	reader, err := s.getReader(readerId)
 	if err != nil {
 		return nil, err
 	}
 
-	return reader.server.GetLatestValue(ctx, req.GetRequest())
+	return reader.server.GetLatestValue(ctx, in)
 }
 
-func (s *Server) ContractReaderGetLatestValueWithHeadData(ctx context.Context, req *relayerset.ContractReaderGetLatestValueRequest) (*pb.GetLatestValueWithHeadDataReply, error) {
-	reader, err := s.getReader(req.ContractReaderId)
+func (s *Server) GetLatestValueWithHeadData(ctx context.Context, in *pb.GetLatestValueRequest) (*pb.GetLatestValueWithHeadDataReply, error) {
+	readerId, err := getContractReaderId(ctx)
+	if err != nil {
+		return nil, err
+	}
+	reader, err := s.getReader(readerId)
 	if err != nil {
 		return nil, err
 	}
 
-	return reader.server.GetLatestValueWithHeadData(ctx, req.GetRequest())
+	return reader.server.GetLatestValueWithHeadData(ctx, in)
 }
 
-func (s *Server) ContractReaderBatchGetLatestValues(ctx context.Context, req *relayerset.ContractReaderBatchGetLatestValuesRequest) (*pb.BatchGetLatestValuesReply, error) {
-	reader, err := s.getReader(req.ContractReaderId)
+func (s *Server) GetLatestValues(ctx context.Context, in *pb.BatchGetLatestValuesRequest) (*pb.BatchGetLatestValuesReply, error) {
+	readerId, err := getContractReaderId(ctx)
+	if err != nil {
+		return nil, err
+	}
+	reader, err := s.getReader(readerId)
```

Comment:
Could combine these:
```suggestion
	reader, err := s.getReader(ctx)
```

Reply:
Done.